examples/PlaneCurve.jl

```diff
@@ -0,0 +1,273 @@
+module PlaneCurveModule
+using Oscar, Markdown
+import Base.:(==)
+
+export Point, ideal_point, AffinePlaneCurve, ProjPlaneCurve, hash, degree,
+       jacobi_ideal, curve_components, isirreducible, isreduced, reduction,
+       union
+
+################################################################################
+
+abstract type PlaneCurve end
+
+################################################################################
+# Point (not specific to curves).
+################################################################################
+# Point (first attempt).
+# The point is described by its coordinates.
+# ambient_dim gives the dimension of the space to which belongs the point.
+
+mutable struct Point{S}
+  coord::Array{S, 1}
+  ambient_dim::Int64

```

Comment:
Unless you really think that you'll need dimensions larger than 2^31, I'd use the "native" int type here and everywhere else instead:
```suggestion
  ambient_dim::Int
```